sktime/utils/validation/_dependencies.py

```diff
@@ -374,6 +375,83 @@ def _check_python_version(obj, package=None, msg=None, severity="error"):
     return True
 
 
+def _check_env_marker(obj, package=None, msg=None, severity="error"):
+    """Check if packaging marker tag is with requirements of obj.
+
+    Parameters
+    ----------
+    obj : sktime estimator, BaseObject descendant
+        used to check python version
+    package : str, default = None
+        if given, will be used in error message as package name
+    msg : str, optional, default = default message (msg below)
+        error message to be returned in the `ModuleNotFoundError`, overrides default
+    severity : str, "error" (default), "warning", or "none"
+        whether the check should raise an error, a warning, or nothing
+
+    Returns
+    -------
+    compatible : bool, whether obj is compatible with system python version
+        check is using the python_version tag of obj
+
+    Raises
+    ------
+    InvalidMarker
+        User friendly error if obj has env_marker tag that is not a
+        packaging compatible marker string
+    ModuleNotFoundError
+        User friendly error if obj has an env_marker tag that is
+        incompatible with the python environment. If package is given,
+        error message gives package as the reason for incompatibility.
+    """
+    est_marker_tag = obj.get_class_tag("env_marker", tag_value_default="None")
+    if est_marker_tag in ["None", None]:
+        return True
+
+    try:
+        est_marker = Marker(est_marker_tag)
+    except InvalidMarker:
+        msg_version = (
+            f"wrong format for env_marker tag, "
+            f'must be PEP 508 compatible specifier string, e.g., os_name!="unix",'
```

Comment:
Ref. https://docs.python.org/3/library/os.html#os.name

I think the only valid values are 'posix', 'nt', 'java'. I think most sktime users/developers (including me, and I'm hoping it's not just me) are likely to be confused by these names and how to map them against windows/macos/unix.

Also, I am not sure if I am doing something wrong, but is it not supposed to fail?

```pycon
>>> 
>>> from packaging.markers import Marker
>>> Marker("os_name == 'blah blah blah'")
<Marker('os_name == "blah blah blah"')>
>>> 
```

Reply:
yes, I see - it would be `platform_system` if you want sth like `"windows"`.

Reply:
> Also, I am not sure if I am doing something wrong, but is it not supposed to fail?

Don't know. `packaging` implements PEP 508 and is the reference implementation, so perhaps it is designed to be upwards compatible, in case someone release a series of operating systems named `"blah blah blah"` later on.